tests/unit_tests/cuda/navigator_cuda.cpp

```diff
@@ -0,0 +1,164 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include <chrono>
+#include <iostream>
+#include <vecmem/memory/cuda/device_memory_resource.hpp>
+#include <vecmem/memory/cuda/managed_memory_resource.hpp>
+
+#include "navigator_cuda_kernel.hpp"
+#include "vecmem/utils/cuda/copy.hpp"
+
+TEST(navigator_cuda, navigator) {
+
+    // Helper object for performing memory copies.
+    vecmem::cuda::copy copy;
+
+    // VecMem memory resource(s)
+    vecmem::cuda::managed_memory_resource mng_mr;
+    vecmem::cuda::device_memory_resource dev_mr;
+
+    // Create detector
+    detector_host_t det =
+        create_toy_geometry<darray, thrust::tuple, vecmem::vector,
+                            vecmem::jagged_vector>(mng_mr, n_brl_layers,
+                                                   n_edc_layers);
+
+    // Create navigator
+    navigator_host_t n(det, mng_mr);
+
+    // Create the vector of initial track parameters
+    vecmem::vector<track<nav_context>> tracks_host(&mng_mr);
+    vecmem::vector<track<nav_context>> tracks_device(&mng_mr);
+
+    // Set origin position of tracks
+    const point3 ori{0., 0., 0.};
+
+    // Loops of theta values ]0,pi[
+    for (unsigned int itheta = 0; itheta < theta_steps; ++itheta) {
+        scalar theta = 0.001 + itheta * (M_PI - 0.001) / theta_steps;
+        scalar sin_theta = std::sin(theta);
+        scalar cos_theta = std::cos(theta);
+
+        // Loops of phi values [-pi, pi]
+        for (unsigned int iphi = 0; iphi < phi_steps; ++iphi) {
+            // The direction
+            scalar phi = -M_PI + iphi * (2 * M_PI) / phi_steps;
+            scalar sin_phi = std::sin(phi);
+            scalar cos_phi = std::cos(phi);
+            const point3 dir{cos_phi * sin_theta, sin_phi * sin_theta,
+                             cos_theta};
+
+            track<nav_context> ray = {.pos = ori,
+                                      .dir = dir,
+                                      .momentum = 100,
+                                      .ctx = nav_context{},
+                                      .overstep_tolerance = -1e-4};
```

Comment:
Isn't this C++20?

Reply:
You're absolutely right, it is!

https://en.cppreference.com/w/cpp/language/aggregate_initialization

It just happens that GCC accepts this formalism even with `-std=gnu++17`. :frowning:

We'll need to set `CMAKE_CXX_EXTENSIONS=FALSE`...

Reply:
This was fixed. Thanks for pointing out